dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/NeuralNetworkParameterServerTask.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2015 Seoul National University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.snu.dolphin.dnn;
+
+import edu.snu.dolphin.bsp.core.DataParser;
+import edu.snu.dolphin.bsp.examples.ml.parameters.MaxIterations;
+import edu.snu.dolphin.dnn.data.NeuralNetParamWorkerData;
+import edu.snu.dolphin.dnn.util.Validator;
+import edu.snu.dolphin.ps.worker.ParameterWorker;
+import org.apache.reef.annotations.audience.TaskSide;
+import org.apache.reef.io.network.util.Pair;
+import org.apache.reef.tang.annotations.Parameter;
+import org.apache.reef.task.Task;
+import org.nd4j.linalg.api.ndarray.INDArray;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static edu.snu.dolphin.dnn.NeuralNetworkTask.*;
+
+/**
+ * The task that trains neural network with the data set.
+ *
+ * Assumes the input file can be parsed by NeuralNetworkDataParser.
+ */
+@TaskSide
+public final class NeuralNetworkParameterServerTask implements Task {
```

Comment:
We named `GroupCommNeuralNetworkTask` for tasks that train neural network model using group communication.
For this class, shouldn't we use a name consistent with `GroupCommNeralNetworkTask` such as `ParameterServerNeuralNetworkTask`?